File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/specificationOptionService.js
app.service('specificationOptionService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../specificationOption/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../specificationOption/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeSpecificationOptionList = function (idList) {
        return $http.get('../specificationOption/removeSpecificationOption.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../specificationOption/findOne.do?id=' + id)
    };

    //修改品牌
    this.updateSpecificationOption = function (specificationOption) {
        return $http.post('../specificationOption/updateSpecificationOption.do', specificationOption)
    };

    //新增品牌
    this.saveSpecificationOption = function (specificationOption) {
        return $http.post('../specificationOption/saveSpecificationOption.do', specificationOption)
    };   
});

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationOptionService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbSpecificationOption;							  
import entity.PageResult;
import entity.Result;
/**
 * specificationOption服务层接口
 * @author jieweili
 *
 */
public interface SpecificationOptionService {

    /**
     * 返回全部列表(已弃置)
     * @return
     */
    List<TbSpecificationOption> findAll();
 
    /**
     * 返回分页对象(过时)
     * @param pageNum
     * @param pageSize
     * @return
     */
	 
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据
     * @param specificationOption
     * @return
     */
    public Result saveSpecificationOption(TbSpecificationOption specificationOption);

    /**
     * 根据id查询一个对象
     * @param id
     * @return
     */
    TbSpecificationOption getSpecificationOptionById(Long id);

    /**
     * 修改对象
     * @param specificationOption
     * @return
     */
    Result updateSpecificationOption(TbSpecificationOption specificationOption);

    /**
     * 批量删除
     * @param idList
     * @return
     */
    Result removeSpecificationOption(Long[] idList);

    /**
     * 根据条件分页查询
     * @param specificationOption
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listSpecificationOptionByExample(TbSpecificationOption specificationOption, Integer pageNum, Integer pageSize);
 
}
